### PR TITLE
helpers/GenerateResourcesAndImage.ps1: check for interactive mode

### PR DESCRIPTION
**Pull request by ilia-shipitsin**
# Description

helpers/GenerateResourcesAndImage.ps1: check for interactive mode

#### Related issue:

https://github.com/actions/runner-images/issues/8298

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: helpers/GenerateResourcesAndImage.ps1 [hunk 1]
```diff
@@ -266,16 +266,23 @@ Function GenerateResourcesAndImage {
                 $ResourceGroupExists = $false
             }
             else {
-                # Resource group already exists, ask the user what to do
-                $title = "Resource group '$ResourceGroupName' already exists"
-                $message = "Do you want to delete the resource group and all resources in it?"
+                # are we running in a non-interactive session?
+                # https://stackoverflow.com/questions/9738535/powershell-test-for-noninteractive-mode
+                if ([System.Console]::IsOutputRedirected -or ![Environment]::UserInteractive -or !!([Environment]::GetCommandLineArgs() | Where-Object { $_ -ilike '-noni*' })) {
```
Comment (Alexey-Ayupov) [inline helpers/GenerateResourcesAndImage.ps1:271]: Do we need double `-not` here -  (**!!**([Environment]::GetCommandLineArgs() | Where-Object { $_ -ilike '-noni*' })))?

Comment (ilia-shipitsin) [reply helpers/GenerateResourcesAndImage.ps1:271]: can you elaborate on some example when it comes to

Comment (Alexey-Ayupov) [reply helpers/GenerateResourcesAndImage.ps1:271]: I got it. It turns the value from expression to the boolean (True or False)

**Review by Alexey-Ayupov — APPROVED** [review]

File: helpers/GenerateResourcesAndImage.ps1 [hunk 2]
```diff
@@ -266,16 +266,23 @@ Function GenerateResourcesAndImage {
...
+                    throw "Non-interactive mode, resource group '$ResourceGroupName' already exists, specify -Force to delete it."
```
Comment (vpolikarpov-akvelon) [inline helpers/GenerateResourcesAndImage.ps1:272]: There is also 'ReuseResourceGroup' flag that is worth mentioning.

File: helpers/GenerateResourcesAndImage.ps1 [hunk 3]
```diff
@@ -266,16 +266,23 @@ Function GenerateResourcesAndImage {
...
+                }
+                else {
```
Comment (vpolikarpov-akvelon) [inline helpers/GenerateResourcesAndImage.ps1:274]: You don't need `else` here because code inside `if` block interrupts execution

Comment (ilia-shipitsin) [reply helpers/GenerateResourcesAndImage.ps1:274]: technically, I agree with you, but may "else" improve readability , it wraps code block which requires interactive mode.

File: helpers/GenerateResourcesAndImage.ps1 [hunk 4]
```diff
@@ -376,4 +383,4 @@ Function GenerateResourcesAndImage {
         }
         Write-Verbose "Cleanup completed."
     }
-}
+}
```
Comment (vpolikarpov-akvelon) [inline helpers/GenerateResourcesAndImage.ps1:386]: Please keep newline at the end of the file

File: helpers/GenerateResourcesAndImage.ps1 (hunk 2 above)
Comment (ilia-shipitsin) [reply helpers/GenerateResourcesAndImage.ps1:272]: makes sense

File: helpers/GenerateResourcesAndImage.ps1 (hunk 4 above)
Comment (ilia-shipitsin) [reply helpers/GenerateResourcesAndImage.ps1:386]: do we have ".vscode" setting for that ?

**Review by mikhailkoliada — APPROVED** [review]

**Pull request merged**